services/web/server/src/simcore_service_webserver/socketio/messages.py

```diff
@@ -47,7 +47,7 @@ async def send_messages(
             sio.emit(
                 message["event_type"],
                 json_dumps(message["data"]),
-                room=SocketIORoom.from_socket_id(sid),
+                room=SocketIORoomStr.from_socket_id(sid),
```

Comment:
I am fine with parsing when only necessary. For that very reason there are the functions like `.construct` or `cast`. That is the responsibility of the developer. 

What i think it is wrong is to avoid annotating correctly. The more precise one is  in the annotation, the more clear the code is and the more you help the static analysis.

In this particular case, I totally understand that `sid` does not need further validation and can simply be set as `SocketIORookStr(sdi)` which does not run a validation.

Reply:
fully agree for the annotation. and also using the ```.construct``` method to reduce the parsing load.

Reply:
OK, let's switch to construct here